File: sdk-android/src/main/java/org/sinou/android/pydia/data/AccountDao.java
package org.sinou.android.pydia.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface AccountDao {

    @Insert
    void insert(Account account);

    @Query("SELECT * FROM account_records ORDER BY login, url ASC ")
    LiveData<List<Account>> listAccounts();

    @Query("DELETE FROM account_records")
    void clearAll();

}

File: sdk-android/src/main/java/org/sinou/android/pydia/data/AccountDao.kt
package org.sinou.android.pydia.data

import androidx.room.*

@Dao
interface AccountDao {


    @Query("SELECT * FROM accounts")
    fun getAllAccounts(): List<Account>
    //fun getAllAccounts(): LiveData<List<Account>>
/*

    @Query("SELECT * FROM accounts WHERE uid IN (:accountIds)")
    fun loadAllByIds(accountIds: IntArray): List<Account>
*/


    //    @Insert
//    suspend fun insertAll(vararg accounts: Account)
    @Insert
    fun insertAll(vararg accounts: Account): List<Long>

    //    @Insert
//    suspend fun insert(account: Account): Int
    @Insert
    fun insert(account: Account): Long

    @Update
    suspend fun update(account: Account)

    @Delete
    fun delete(account: Account)
}